src/test/java/io/jenkins/plugins/analysis/core/model/ByIdResultSelectorTest.java

```diff
@@ -0,0 +1,64 @@
+package io.jenkins.plugins.analysis.core.model;
+
+import hudson.model.Run;
+import io.jenkins.plugins.analysis.core.views.ResultAction;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ByIdResultSelectorTest {
+
+	private ByIdResultSelector cut;
+
+	@Test
+	void shouldBeEmptyOptionalFromEmptyList() {
+		cut  = new ByIdResultSelector("1");
+		Run<?, ?> run = mock(Run.class);
+		when(run.getActions(ResultAction.class)).thenReturn(new ArrayList<>());
+
+		Optional<ResultAction> resultAction = cut.get(run);
+		assertThat(resultAction).isEmpty();
+	}
+
+	@Test
+	void shouldBeEmptyOptionalFromList() {
+		cut  = new ByIdResultSelector("1");
+		ResultAction resultAction = mock(ResultAction.class);
+		when(resultAction.getId()).thenReturn("2");
+
+		List<ResultAction> resultActions = new ArrayList<>();
+		resultActions.add(resultAction);
+
+		Run<?, ?> run = mock(Run.class);
+		when(run.getActions(ResultAction.class)).thenReturn(resultActions);
+
+		assertThat(cut.get(run)).isEmpty();
+	}
+
+	@Test
+	void shouldBeNonEmptyOptional() {
+		cut  = new ByIdResultSelector("1");
+		ResultAction resultAction = mock(ResultAction.class);
+		when(resultAction.getId()).thenReturn("1");
+
+		List<ResultAction> resultActions = new ArrayList<>();
+		resultActions.add(resultAction);
+
+		Run<?, ?> run = mock(Run.class);
+		when(run.getActions(ResultAction.class)).thenReturn(resultActions);
+
+		assertThat(cut.get(run)).isPresent();
+	}
+
```

Comment:
In order to get Boundary Interior Path Coverage of 100% there should be another two tests:
- List = {Action("2"), Action("3")} 
- List = {Action("2"), Action("1")}